nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -170,7 +174,9 @@ WaypointFollower::followWaypoints()
   }
 
   rclcpp::WallRate r(loop_rate_);
-  uint32_t goal_index = 0;
+
+  // get the goal index, by default, the first in the list of waypoints given.
+  uint32_t goal_index = goal->goal_index;
```

Comment:
That was the discussion which we apparently had few months ago. We do need this to recover the waypoint where the goal was paused..

Reply:
That's preempted with a new request while the current request is active. It could be from an entirely new set of goals - why would we ignore one of the fields for a new request? This isn't just for the RViz plugin tool if its in this server

Reply:
Yes you are right, we just send a completely new request after we pause and restart the waypoint following. During pausing we just store the index of the goal to which the robot was navigating.

Reply:
But when you pause, I think you actually cancel and then reissue later the WPF command? That's fine with how things work right now. The change I'm requesting is for preemption, which is not currently exercised by that workflow

Reply:
Got it! You meant L197! My bad, I some was stuck in L179!